test/slow.js

```diff
@@ -132,6 +132,29 @@ void (async() => {
       'Accept-Encoding': 'gzip, deflate, br',
     },
     body: '{}',
+  }, {
+    url: 'https://dynamodb.us-east-1.amazonaws.com/',
+    headers: {
+      'Content-Type': 'application/x-amz-json-1.0',
+      'X-Amz-Target': 'DynamoDB_20120810.ListTables',
+      'Accept-Encoding': 'gzip, deflate, br',
+      'User-Agent': 'node',
+    },
+    extraHeadersToInclude: {
+      'user-agent': true,
+    },
+    body: '{}',
+  }, {
+    url: 'https://dynamodb.us-east-1.amazonaws.com/',
+    headers: {
+      'Content-Type': 'application/x-amz-json-1.0',
+      'X-Amz-Target': 'DynamoDB_20120810.ListTables',
+      'Accept-Encoding': 'gzip, deflate, br',
+    },
+    extraHeadersToIgnore: {
+      'content-type': true,
+    },
+    body: '{}',
```

Comment:
These new integration tests cover regressions for this bugfix, since they fail without the accompanying changes in `aws4`.

_This assumes the `npm run integration` command is run with AWS credentials which grant sufficient access (for example, the [`ReadOnlyAccess` AWS-managed policy](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/ReadOnlyAccess.html))._

```
$ npm run integration

> aws4@1.13.0 integration
> node ./test/slow.js

Passed tests 0
Passed tests 1
Passed tests 2
Passed tests 3
Passed tests 4
Passed tests 5
Passed tests 6
Passed tests 7
Passed tests 8
Passed tests 9
Passed tests 10
Passed tests 11
Passed tests 12
Passed tests 13
Passed tests 14
Passed tests 15
Passed tests 16
Passed tests 17
Passed tests 18
Passed tests 19
Passed tests 20
Passed tests 21
Passed tests 22
Passed tests 23
Passed tests 24
Passed tests 25
Passed tests 26
Passed tests 27
Passed tests 28
Passed tests 29
Passed tests 30
Passed tests 31
Passed tests 32
Passed tests 33
Passed tests 34
Passed tests 35
Passed tests 36
Passed tests 37
Passed tests 38
Passed tests 39
Passed tests 40
Passed tests 41
Passed tests 42
Passed tests 43
Passed tests 44
Passed tests 45
Passed tests 46
Passed tests 47
Passed tests 48
Passed tests 49
Passed tests 50
Passed tests 51
Passed tests 52
Passed tests 53
Passed tests 54
Passed tests 55
Passed tests 56
Passed tests 57
Passed tests 58
Passed tests 59
Passed tests 60
Passed tests 61
Passed tests 62
Passed tests 63
Passed tests 64
Passed tests 65
Passed tests 66
Passed tests 67
Passed tests 68
Passed tests 69
Passed tests 70
Passed tests 71
Passed tests 72
Passed tests 73
Passed tests 74
Passed tests 75
Passed tests 76
Passed tests 77
Passed tests 78
Passed tests 79
Passed okTests 0
Passed okTests 1
Passed okTests 2
Passed okTests 3
Passed okTests 4
Passed okTests 5
Passed okTests 6
Passed okTests 7
Passed okTests 9
Passed okTests 10
Passed okTests 11
Passed okTests 12
Passed okTests 13
Passed okTests 14
Passed okTests 15
Passed okTests 16
Passed okTests 17
Passed okTests 18
Passed okTests 19
Passed okTests 20
Passed okTests 21
Passed okTests 22
Passed okTests 23
Passed okTests 24
Passed okTests 25
Passed okTests 26
Passed okTests 27
Passed okTests 28
Passed okTests 29
Passed okTests 30
Passed okTests 31
Passed okTests 32
Passed okTests 33
Passed okTests 34
Passed okTests 35
Passed okTests 36
Passed okTests 37
Passed okTests 38
Passed okTests 39
Passed okTests 40
Passed okTests 41
Passed okTests 42
Passed okTests 43
Passed okTests 44
Passed okTests 45
Passed okTests 46
Passed okTests 47
Passed okTests 48
Passed okTests 49
Passed okTests 50
Passed okTests 52
Passed okTests 53
Passed okTests 54
Passed okTests 55
Passed okTests 56
Passed okTests 57
Passed okTests 58
Passed okTests 59
Passed okTests 60
Passed okTests 61
Passed okTests 62
Passed okTests 63
Passed okTests 64
Passed okTests 65
Passed okTests 66
Passed okTests 67
Passed okTests 68
Passed okTests 69
Passed okTests 70
Passed okTests 71
Passed okTests 72
Passed okTests 73
Passed okTests 74
Passed okTests 75
Passed okTests 76
Passed okTests 77
Passed okTests 78
Passed okTests 79
Passed okTests 80
Passed okTests 81
Passed okTests 82
Passed okTests 83
Passed okTests 84
Passed okTests 85
Test 8
GET /?Action=ListDomainNames&Version=2013-01-01
--------------
https://cloudsearch.us-east-1.amazonaws.com/?Action=ListDomainNames&Version=2013-01-01
--------------
GET
/
Action=ListDomainNames&Version=2013-01-01
host:cloudsearch.us-east-1.amazonaws.com
x-amz-date:20240725T200912Z
x-amz-security-token:REDACTED

host;x-amz-date;x-amz-security-token
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
--------------
<ErrorResponse xmlns="http://cloudsearch.amazonaws.com/doc/2013-01-01/">
  <Error>
    <Type>Sender</Type>
    <Code>NotAuthorized</Code>
    <Message>New domain creation not supported on this account. Please reach out to AWS Support for assistance.</Message>
  </Error>
  <RequestId>4153b6d6-8adf-430a-9fbe-82f6feb74ba7</RequestId>
</ErrorResponse>

==============
Test 51
GET /?Action=ListDomainNames&Version=2013-01-01&X-Amz-Security-Token=REDACTED&X-Amz-Date=20240725T200912Z&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=REDACTED%2F20240725%2Fus-east-1%2Fcloudsearch%2Faws4_request&X-Amz-SignedHeaders=host&X-Amz-Signature=92b28ea9eac1877f751cc6a24cf7a6f1c64320187dc8d497f534e109d884488e
--------------
https://cloudsearch.us-east-1.amazonaws.com/?Action=ListDomainNames&Version=2013-01-01
--------------
GET
/
Action=ListDomainNames&Version=2013-01-01&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=REDACTED%2F20240725%2Fus-east-1%2Fcloudsearch%2Faws4_request&X-Amz-Date=20240725T200912Z&X-Amz-Security-Token=REDACTED&X-Amz-Signature=92b28ea9eac1877f751cc6a24cf7a6f1c64320187dc8d497f534e109d884488e&X-Amz-SignedHeaders=host
host:cloudsearch.us-east-1.amazonaws.com

host
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
--------------
<ErrorResponse xmlns="http://cloudsearch.amazonaws.com/doc/2013-01-01/">
  <Error>
    <Type>Sender</Type>
    <Code>NotAuthorized</Code>
    <Message>New domain creation not supported on this account. Please reach out to AWS Support for assistance.</Message>
  </Error>
  <RequestId>6789d12a-b613-4a44-b4ac-428127bb533b</RequestId>
</ErrorResponse>

==============
```